src/views/datasources/actions/DeleteDataSourceModal/DeleteDataSourceModal.tsx

```diff
@@ -0,0 +1,103 @@
+import React, { FC, useState } from 'react';
+import { Trans } from 'react-i18next';
+
+import { PersistentVolumeClaimModel } from '@kubevirt-ui/kubevirt-api/console';
+import DataImportCronModel from '@kubevirt-ui/kubevirt-api/console/models/DataImportCronModel';
+import DataSourceModel from '@kubevirt-ui/kubevirt-api/console/models/DataSourceModel';
+import DataVolumeModel from '@kubevirt-ui/kubevirt-api/console/models/DataVolumeModel';
+import {
+  V1beta1DataImportCron,
+  V1beta1DataSource,
+} from '@kubevirt-ui/kubevirt-api/containerized-data-importer/models';
+import ConfirmActionMessage from '@kubevirt-utils/components/ConfirmActionMessage/ConfirmActionMessage';
+import TabModal from '@kubevirt-utils/components/TabModal/TabModal';
+import { useKubevirtTranslation } from '@kubevirt-utils/hooks/useKubevirtTranslation';
+import { getName, getNamespace } from '@kubevirt-utils/resources/shared';
+import { kubevirtConsole } from '@kubevirt-utils/utils/utils';
+import { k8sDelete } from '@openshift-console/dynamic-plugin-sdk';
+import { ButtonVariant, Checkbox, Stack, StackItem } from '@patternfly/react-core';
+
+import useUnderlyingPVC from './hooks/useUnderlyingPVC';
+
+type DeleteDataSourceModalProps = {
+  dataImportCron: V1beta1DataImportCron;
+  dataSource: V1beta1DataSource;
+  isOpen: boolean;
+  onClose: () => void;
+};
+
+const DeleteDataSourceModal: FC<DeleteDataSourceModalProps> = ({
+  dataImportCron,
+  dataSource,
+  isOpen,
+  onClose,
+}) => {
+  const { t } = useKubevirtTranslation();
+
+  const [deletePVC, setDeletePVC] = useState<boolean>(false);
+  const { dv, pvc } = useUnderlyingPVC(dataSource);
+
+  const onDelete = async () => {
+    try {
+      await k8sDelete({
+        model: DataImportCronModel,
+        resource: dataImportCron,
+      });
+    } catch (e) {
+      kubevirtConsole.log(e?.message);
+    } finally {
+      await k8sDelete({
+        model: DataSourceModel,
+        resource: dataSource,
+      });
```

Comment:
this wont throw?

Reply:
I can wrap this in try catch and handle the error, wdyt?

Reply:
i see from the code that you always want to delete dataSource , if u will change the order?
first, delete dataSource then dataimportcron ? is possible? then u will need only one try catch

Reply:
Actually, this code block was taken from the DS actions hook (not copied) and added the part where I delete the PVC and DV, Since we're a bit lacking time, how about I check that as a follow-up task?

Reply:
ok but please open a ticket on u so it wont be forgotten :)

Reply:
@metalice I've checked this subject, We are deleting the DIC first because the DIC CR is auto-populating the DS if not exist. Adding this image of my terminal (hopefully it will help clarify this topic):
![Screenshot from 2023-09-12 11-16-07](https://github.com/kubevirt-ui/kubevirt-plugin/assets/67270715/6fde2769-e786-4d7a-9086-a08ad2154a2b)

So I guess that's the we should add them together to the same try block, instead of finally that will run even if the DIC deletion fails (which will auto-heal the DS deletion)